LanguageFeatures/Extension-types/exhaustiveness_object_A01_t04.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with a sealed class as a
+/// matched type are always exhaustive
+///
+/// @description Check that it is no compile-time error if the matched value
+/// type of a switch statement is a sealed class as and the set of cases is an
+/// exhaustive set of object patterns with extension types
```

Comment:
This one is again blocked.

Reply:
Unblocked now? No error in the current implementation

Reply:
Actually, it was probably not blocked after all.

Anyway, the analysis today says: `Face` is always-exhaustive, so we must check exhaustiveness of switch statements (and switch expressions, too, as always). The extension-typed object patterns are checked during exhaustiveness analysis according to their erasure, and we do have all immediate subtypes of `Face` (with some extra tricks around type arguments).

So: Unblocked indeed.